yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -540,6 +561,29 @@ private[spark] class Client(
     amContainer
   }
 
+  def setupCredentials(): Unit = {
+    Option(args.principal) match {
```

Comment:
I don't know, the nested matches here make my head hurt. Maybe good old if..else..if would be easier to follow.